### PR TITLE
Autonomous

### PR DESCRIPTION
**Pull request by kirbt**
Autonomous Logic
Practice Bot Auto Configs
Pathplanner Paths

### CODE REVIEW

File: .OutlineViewer/outlineviewer.json
Comment (TAKBS2412) [inline .OutlineViewer/outlineviewer.json:1]: Please revert

File: build.gradle
Comment (TAKBS2412) [inline build.gradle:1]: Could we test with 2024.2.1 again to see if we really need to downgrade?

File: bonkswerve.chor
Comment (TAKBS2412) [inline bonkswerve.chor:1]: Do we still need the choreo files, or can we delete them?

File: src/main/deploy/practiceswerve/modules/physicalproperties.json [hunk 1]
```diff
@@ -6,8 +6,8 @@
     "angle": 20
   },
   "conversionFactor": {
-    "angle": 19.2,
-    "drive": 0.059108484
+    "angle": 28.125,
+    "drive": 0.040279076
```
Comment (TAKBS2412) [inline src/main/deploy/practiceswerve/modules/physicalproperties.json]: This seems to conflict with the changes made in #56 (see [here](https://github.com/robototes/Crescendo2024/pull/56/files#diff-52add4cc008e271f6ea2329c782841a0083933b53d4d7b5457680d246d06b7d2)) - which one is correct?

File: src/main/java/frc/team2412/robot/util/AutoLogic.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+public class AutoLogic {
+	public Robot r = Robot.getInstance();
+	private final Subsystems s;
+	private final Controls controls;
+
+	private SequentialCommandGroup vibrateControllerCommand;
+
+	public boolean dummyLogic() {
+		return true;
+	}
```
Comment (TAKBS2412) [inline src/main/java/frc/team2412/robot/util/AutoLogic.java]: What's this method for?

File: src/main/java/frc/team2412/robot/util/ChoreoHandler.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+	// pid
+
+	private static final PIDController xController = new PIDController(0.1, 0, 0);
+	private static final PIDController yController = new PIDController(0.1, 0, 0);
+	private static final PIDController rotationController = new PIDController(0.5, 0, 0);
+
+	// constructor
+
+	public ChoreoHandler() {}
```
Comment (TAKBS2412) [inline src/main/java/frc/team2412/robot/util/ChoreoHandler.java]: Please delete

File: src/main/java/frc/team2412/robot/util/ChoreoHandler.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+	// methods
+
+	private static boolean mirrorTrajectory() {
+		return DriverStation.getAlliance().get() == Alliance.Blue;
```
Comment (TAKBS2412) [inline src/main/java/frc/team2412/robot/util/ChoreoHandler.java]: Have we tested this? IIRC the paths are created on the blue side, so we'd need to flip it when on the red alliance.

**Review by TAKBS2412 — CHANGES_REQUESTED** [review]

File: build.gradle
Comment (kirbt) [reply build.gradle:1]: Will do on thursday

File: src/main/java/frc/team2412/robot/util/AutoLogic.java (hunk 2 above)
Comment (kirbt) [reply src/main/java/frc/team2412/robot/util/AutoLogic.java]: placeholder while waiting for limelight pr (need method for determining if note is present where were looking at)

File: src/main/java/frc/team2412/robot/util/ChoreoHandler.java (hunk 4 above)
Comment (kirbt) [reply src/main/java/frc/team2412/robot/util/ChoreoHandler.java]: oh true

File: bonkswerve.chor
Comment (kirbt) [reply bonkswerve.chor:1]: I will probably delete them

File: src/main/deploy/pathplanner/New Path.path [hunk 5]
```diff
@@ -0,0 +1,74 @@
+{
```
Comment (TAKBS2412) [inline src/main/deploy/pathplanner/New Path.path]: Should delete or rename to be more descriptive.

File: .OutlineViewer/outlineviewer.json
Comment (TAKBS2412) [reply .OutlineViewer/outlineviewer.json:1]: The file is still showing up on git, but as being empty.

File: src/main/deploy/pathplanner/New Path.path (hunk 5 above)
Comment (stewluj) [reply src/main/deploy/pathplanner/New Path.path]: i have no idea when i made this but its not needed so deleted

**Review by TAKBS2412 — COMMENTED** [review]
This looks fine but I think we should test it on the real robot before merging.

File: src/main/java/frc/team2412/robot/Controls.java [hunk 6]
```diff
@@ -156,7 +156,6 @@ private void bindLauncherControls() {
 	}
 
 	public void vibrateDriveController(double vibration) {
-		// no reason to rumble in auto when no one is holding the controller
```
Comment (jamesdooley4) [inline src/main/java/frc/team2412/robot/Controls.java:159]: Was it intended to delete this? The comment seems useful to me

**Review by jamesdooley4 — APPROVED** [review]

**Pull request merged**